cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -109,13 +111,32 @@ func NewCmdApply(apf *applyPlanFlags) *cobra.Command {
 func runApply(flags *applyFlags, userVersion string) error {
 
 	// Start with the basics, verify that the cluster is healthy and get the configuration from the cluster (using the ConfigMap)
-	klog.V(1).Infoln("[upgrade/apply] verifying health of cluster")
+
 	klog.V(1).Infoln("[upgrade/apply] retrieving configuration from cluster")
-	client, versionGetter, cfg, err := enforceRequirements(flags.applyPlanFlags, flags.dryRun, userVersion)
```

Comment:
This was the main problem. the function `enforceRequirements` performs a few things. Not only does it get things we need for upgrading (what will become "ApplyData") So the main goal is to split getting stuff and `enforcing` things and sets us up for the other PR's.